### PR TITLE
Sonnet V1.2.2

### PR DESCRIPTION
**Pull request by ultrabear**
PR for Sonnet V1.2.2

### CODE REVIEW

**Comment by ultrabear** [comment]
# Changelog:
## Frontend:
- Add stricter input validation to set-antispam
- Add set-mutetime command
- unban and unmute now insert reasons and logs into database
- note infractions are like warns but dont log to infraction log or dm the user
- Infraction searching now supports negative indexing
- rr-purge added to remove all reactionroles from a message
- map now takes -e parameter for appending args to the end of every command
- Add absolute totals to % totals in statistics command
- Support floating point seconds in antispam time
## Under the Hood:
- Added return state to most commands, allows sonnetsh and map to stop execution after a failure
- Remove unneeded imports from all files, this is thanks to the new addition of pyflakes and pylint in the workflow
- GDPR now instancizes commands before running them
- General infraction processor is no longer garbage
  - now gaurantees a user object or failure
  - can specify to not add an infraction to logs but put it in the database
  - just look at it its so much cleaner
- muting a user now first clears them from the mute database and unmuting will also clear them from the database
- Ping function now uses timestamp generation template function
- Uptime printing now uses shared function, no longer prints as "days" when the count of days is 1
- Userupdate no longer checks for changed usernames as that event happens on on-user-update
- Update kernel to LeXdPyK 1.3.1
  - Updates to support discord.py 1.7 deprecations
  - error printing is now pretty
## Runtime Environment:
- Make re2 optional
- Make loading the C lib optional
- Add default lite config for running the bot without mariadb or re2
- Move requirements.txt to requirements.txt and requirements.extras.txt to allow for not installing re2 or mariadb
## Bugs:
- Fix jkb on a non user would print None
- Fix negative mute times that would proceed to not mute
- Infraction searching no longer goes to last page when 0 is passed into it

File: libs/lib_loaders.py [hunk 1]
```diff
@@ -53,7 +61,11 @@ def directBinNumber(inData, length):
 
 defaultcache = {
     "csv": [["word-blacklist", ""], ["filetype-blacklist", ""], ["word-in-word-blacklist", ""], ["antispam", "3,2"]],
-    "text": [["prefix", GLOBAL_PREFIX], ["blacklist-action", BLACKLIST_ACTION], ["blacklist-whitelist", ""], ["regex-notifier-log", ""], ["admin-role", ""], ["moderator-role", ""]],
+    "text":
+        [
+            ["prefix", GLOBAL_PREFIX], ["blacklist-action", BLACKLIST_ACTION], ["blacklist-whitelist", ""], ["regex-notifier-log", ""], ["admin-role", ""], ["moderator-role", ""],
+            ["antispam-time", "20"]
+            ],
```
Comment (Saragon4005) [inline libs/lib_loaders.py:68]: thanks yapf

File: requirements.txt [hunk 2]
```diff
@@ -1,5 +1,3 @@
-discord.py >= 1.6
-mariadb >= 1.0.5
+discord.py >= 1.7
```
Comment (Saragon4005) [inline requirements.txt:1]:
```suggestion
discord.py >= 1.7.1
```
patch has been added

**Review by Saragon4005 — CHANGES_REQUESTED** [review]

File: cmds/cmd_moderation.py [hunk 3]
```diff
@@ -95,257 +86,246 @@ async def process_infraction(message, args, client, infraction_type, pretty_infr
...
 
-    if kwargs["verbose"]: await message.channel.send(f"Kicked {user.mention} with ID {user.id} for {reason}"[:2000])
+    if kwargs["verbose"]: await message.channel.send(f"Kicked {member.mention} with ID {member.id} for {reason}", allowed_mentions=discord.AllowedMentions.none())
 
 
 async def ban_user(message, args, client, **kwargs):
 
     try:
-        user, reason, infractionID, is_member, dm_sent = await process_infraction(message, args, client, "ban", "Banning")
+        member, user, reason, _, dm_sent = await process_infraction(message, args, client, "ban")
```
[the remark this answers is not in the document]
Comment (ultrabear) [reply cmds/cmd_moderation.py:177]: ah yes and throw away the feature of banning someone not in the guild? nah

File: cmds/cmd_moderation.py [hunk 4]
```diff
@@ -30,52 +30,43 @@ async def catch_dm_error(user, contents, log_channel):
...
+        db.add_infraction(generated_id, user.id, moderator_id, infraction_type, infraction_reason, round(time.time()))
 
-        # Generate log channel object
-        if channel_id:  # If ID exists then use it
-            log_channel = client.get_channel(int(channel_id))
-        else:
-            log_channel = None
-            send_message = False
+    if not to_dm:
+        return (None, None)
```
[the remark this answers is not in the document]
Comment (ultrabear) [reply cmds/cmd_moderation.py:48]: yeah

File: cmds/cmd_automod.py [hunk 5]
```diff
@@ -320,6 +362,14 @@ async def antispam_set(message, args, client, **kwargs):
             'cache': 'regenerate',
             'execute': antispam_set
             },
+    'set-mutetime':
+        {
+            'pretty_name': 'set-mutetime <time>',
+            'description': 'Set how long a person should be muted for with antispam automute',
```
[the remark this answers is not in the document]
Comment (ultrabear) [reply cmds/cmd_automod.py]: fixed in 91d79f1b1355d06b7aa807d84f016e7685046311

File: cmds/cmd_automod.py [hunk 6]
```diff
@@ -218,23 +223,60 @@ async def antispam_set(message, args, client, **kwargs):
...
+        return 1
+    elif seconds < 0:
+        await message.channel.send("ERROR: Cannot go below 0 seconds")
+        return 1
 
     with db_hlapi(message.guild.id) as database:
-        database.add_config("antispam", f"{messages},{seconds}")
+        database.add_config("antispam", f"{int(messages)},{seconds}")
+
+    if kwargs["verbose"]: await message.channel.send(f"Updated antispam timings to M:{int(messages)},S:{seconds}")
```
[the remark this answers is not in the document]
Comment (ultrabear) [reply cmds/cmd_automod.py:253]: fixed in 91d79f1b1355d06b7aa807d84f016e7685046311

**Review by Saragon4005 — APPROVED** [review]

**Pull request merged**